website_hr_contact/controllers/main.py

```diff
@@ -0,0 +1,135 @@
+    _items_per_page = 20
+
+    def _get_children_department_recursive(self, root, visible_departments):
+        tree = []
+        for department in root.child_ids:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
+                        lambda rec: rec.id == department.id),
+                    'children': children
+                    })
+            else:
+                tree.extend(children)
+        return tree
+
+    def _get_departments_tree(self, department=None, search=None, **post):
+        """ Build a department tree with only published departments
+        The tree only contains published department even if the parent
+        department is not published.
+        """
+        tree = []
+        request = http.request
+        hr_department = request.env['hr.department']
+        sudo_hr_department = hr_department.sudo()
+        visible_departments = hr_department.search([(1, '=', 1)])
+        departments = sudo_hr_department.search([('parent_id', '=', False)])
+        for department in departments:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
+                        lambda rec: rec.id == department.id),
+                    'children': children
+                    })
+            else:
+                tree.extend(children)
+        return tree
+
+    def _get_employees_domain(self, department=None, search=None, **post):
+        domain = []
+        if department:
+            domain.append(('department_id', '=', department.id))
+        if search:
+            domain.append(('name', 'ilike', search))
+        return domain
+
+    @http.route(['/page/hr_contact/employees',
+                 '/page/hr_contact/employees/page/<int:page>',
+                 '/page/hr_contact/employees/department/'
+                 '<model("hr.department"):department>',
+                 '/page/hr_contact/employees/department/'
+                 '<model("hr.department"):department>/page/<int:page>'
+                 ], type='http', auth="public", website=True)
+    def employees(self, department=None, search='', page=1, **post):
+        request = http.request
+        hr_employee = request.env['hr.employee']
+        departments_tree = self._get_departments_tree(
+            department, search, **post)
+
+        domain = self._get_employees_domain(department, search, **post)
+        employees = hr_employee.search(domain)
+
+        url_builder = QueryURL(
+            '/page/hr_contact/employees', ['department', 'page'],
+            department=department)
+
+        count = len(employees)
```

Comment:
can't you use search_count and then use limit+offeset to limit the load of the search?